web/src/server/api/routers/annotationQueues.ts

```diff
@@ -0,0 +1,532 @@
+import { auditLog } from "@/src/features/audit-logs/auditLog";
+import { throwIfNoEntitlement } from "@/src/features/entitlements/server/hasEntitlement";
+import { throwIfNoProjectAccess } from "@/src/features/rbac/utils/checkProjectAccess";
+import {
+  createTRPCRouter,
+  protectedProjectProcedure,
+} from "@/src/server/api/trpc";
+import {
+  AnnotationQueueObjectType,
+  AnnotationQueueStatus,
+  CreateQueueData,
+  filterAndValidateDbScoreConfigList,
+  LangfuseNotFoundError,
+  optionalPaginationZod,
+  Prisma,
+} from "@langfuse/shared";
+import { logger } from "@langfuse/shared/src/server";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+
+export const queueRouter = createTRPCRouter({
+  all: protectedProjectProcedure
+    .input(
+      z.object({
+        projectId: z.string(),
+        ...optionalPaginationZod,
+      }),
+    )
+    .query(async ({ input, ctx }) => {
+      try {
+        throwIfNoEntitlement({
+          entitlement: "annotation-queues",
+          projectId: input.projectId,
+          sessionUser: ctx.session.user,
+        });
+
+        throwIfNoProjectAccess({
+          session: ctx.session,
+          projectId: input.projectId,
+          scope: "annotationQueues:read",
+        });
+
+        const [queues, totalCount, scoreConfigs] = await Promise.all([
+          ctx.prisma.$queryRaw<
+            Array<{
+              id: string;
+              name: string;
+              description?: string | null;
+              scoreConfigs: string[];
+              createdAt: string;
+              countCompletedItems: number;
+              countPendingItems: number;
+            }>
+          >(Prisma.sql`
+          SELECT
+            aq.id,
+            aq.name,
+            aq.description,
+            aq.score_configs AS "scoreConfigs",
+            aq.created_at AS "createdAt",
+            COALESCE(SUM(CASE WHEN aqi.status = 'COMPLETED' THEN 1 ELSE 0 END), 0) AS "countCompletedItems",
+            COALESCE(SUM(CASE WHEN aqi.status = 'PENDING' THEN 1 ELSE 0 END), 0) AS "countPendingItems"
+          FROM
+            annotation_queues aq
+          LEFT JOIN
+            annotation_queue_items aqi ON aq.id = aqi.queue_id AND aqi.project_id = ${input.projectId}
```

Comment:
i assume it is a bit more efficient to have the project_id condition like aqi.project_id = aq.project_id.